### DISCUSSION TITLE
Filtering by resource pool, vAPP ...

### DISCUSSION THREAD

**Discussion by BobZteD** [bb-Ricardo/netbox-sync#309] [Q&A]
Hi and thanks for this amazing script that saved me lotssss of time
I want to know is there any way to filter importing VMs by Resource pool or vApps? I tried  to use  cluster filter for this but it didn't worked:
cluster_exclude_filter = "CS-UCS/~RnD"
and if using custom attribute works, please help how to use them.

may your RAM be utilized less than 80% all the time ;)

**Comment by bb-Ricardo** [comment]
Hey @BobZteD,

the `cluster_exclude_filter` should work. This field uses RegEx strings, therefore the `/` needs to be escaped.
try (https://regex101.com/r/XyRE6J/1):
```ini
cluster_exclude_filter = "CS-UCS\/~RnD"
```

**Reply by BobZteD** [reply]
Thanks, I'll try

**Reply by BobZteD** [reply]
I have tried in many ways but no success , I think it did not pick this regex for vapp or resource pool although they addressing looks similar
I've imported whole and removed unnecessary vms afterward. 
Regards

**Reply by bb-Ricardo** [reply]
Hi,

Exclusion of VMs only works on cluster and vm level. The tool is not aware of vapps or resource pools. Not sure how they are exposed via API.

**Reply by BobZteD** [reply]
Since I know nothing about programming, take a look at your codes and find [connection.py](https://github.com/bb-Ricardo/netbox-sync/blob/main/module/sources/vmware/connection.py) and from this `vim.ClusterComputeResourc` find this [vim.ResourcePool](https://vdc-download.vmware.com/vmwb-repository/dcr-public/3325c370-b58c-4799-99ff-58ae3baac1bd/45789cc5-aba1-48bc-a320-5e35142b50af/doc/vim.ResourcePool.html) I hope this is usefull
